### ISSUE TITLE
CHORD算法在Qwen2.5-VL-32B不work

### ISSUE THREAD

**Issue by ycp1027tencent** [modelscope/ms-swift#6358]
grpo与cold start数据格式完全相同，json文件包含images、problem、answer、solution四个字段。cold start中带有think过程，grpo只有标签。但log中感觉sft过程没有学习，配置文件信息如下：

export LOG_LEVEL=DEBUG
MAX_PIXELS=200704 \
WANDB_API_KEY=97c188294999d80264588d15d62791cf5af3da6a \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=8 \
swift rlhf \
    --rlhf_type grpo \
    --model /group/40106/model_output/Qwen_96w_full/checkpoint-227652 \
    --model_type qwen2_5_vl \
    --dataset /group/40106/data/antifraud_lvlm/rl_data/data/train_with_solution.json \
    --val_dataset /group/40106/data/antifraud_lvlm/rl_data/data/val_with_solution.json \
    --load_from_cache_file True \
    --torch_dtype bfloat16 \
    --beta 0.0 \
    --lora_rank 64 \
    --max_pixels 200704 \
    --generation_batch_size 32 \
    --num_train_epochs 1 \
    --max_length 4096 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 2\
    --chord_sft_per_device_train_batch_size 4 \
    --chord_sft_dataset /group/40106/data/antifraud_lvlm/cold_start/data/cold_start_chord_with_solution.json \
    --chord_enable_phi_function True \
    --chord_mu_warmup_steps 0 \
    --chord_mu_decay_steps 1000 \
    --chord_mu_peak 0.95 \
    --chord_mu_valley 0.05 \
    --num_generations 8 \
    --train_type lora \
    --freeze_vit true \
    --attn_impl flash_attention_2 \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --reward_funcs antifraud_reward \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.5 \
    --vllm_max_model_len 8192 \
    --vllm_max_lora_rank 64 \
    --vllm_tensor_parallel_size 4 \
    --max_completion_length 1024 \
    --overlong_filter true \
    --offload_optimizer true \
    --offload_model true \
    --sleep_level 1 \
    --save_steps 1000 \
    --eval_steps 1000 \
    --learning_rate 1e-6 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --deepspeed zero3 \
    --move_model_batches 20 \
    --use_liger_kernel false \
    --output_dir /group/40106/model_output_rl/qwen25_chord \
    --log_completions true

**Comment by ycp1027tencent** [comment]
`{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.3125, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08832435, "global_step/max_steps": "1196/13541", "percentage": "8.83%", "elapsed_time": "13h 8m 40s", "remaining_time": "5d 15h 40m 42s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025274}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.3125, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.0883982, "global_step/max_steps": "1197/13541", "percentage": "8.84%", "elapsed_time": "13h 9m 16s", "remaining_time": "5d 15h 39m 20s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025276}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.25, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08847205, "global_step/max_steps": "1198/13541", "percentage": "8.85%", "elapsed_time": "13h 9m 47s", "remaining_time": "5d 15h 37m 14s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025281}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.25, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.0885459, "global_step/max_steps": "1199/13541", "percentage": "8.85%", "elapsed_time": "13h 10m 31s", "remaining_time": "5d 15h 37m 19s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025279}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.21875, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08861975, "global_step/max_steps": "1200/13541", "percentage": "8.86%", "elapsed_time": "13h 11m 11s", "remaining_time": "5d 15h 36m 42s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025278}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.125, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.0886936, "global_step/max_steps": "1201/13541", "percentage": "8.87%", "elapsed_time": "13h 11m 47s", "remaining_time": "5d 15h 35m 28s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.02528}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.71875, "completions/min_length": 2.0, "completions/max_length": 7.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08876745, "global_step/max_steps": "1202/13541", "percentage": "8.88%", "elapsed_time": "13h 12m 24s", "remaining_time": "5d 15h 34m 22s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025282}
{"loss": 0.0, "grad_norm": 0.04959266, "learning_rate": 1e-06, "reward": -4.9375, "reward_std": 0.17677669, "frac_reward_zero_std": 0.75, "rewards/AntifraudReward/mean": -4.9375, "rewards/AntifraudReward/std": 0.35355338, "completions/mean_length": 9.5625, "completions/min_length": 2.0, "completions/max_length": 204.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.0888413, "global_step/max_steps": "1203/13541", "percentage": "8.88%", "elapsed_time": "13h 13m 2s", "remaining_time": "5d 15h 33m 24s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025283}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.125, "completions/min_length": 2.0, "completions/max_length": 4.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08891515, "global_step/max_steps": "1204/13541", "percentage": "8.89%", "elapsed_time": "13h 13m 38s", "remaining_time": "5d 15h 32m 12s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025284}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.84375, "completions/min_length": 2.0, "completions/max_length": 5.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.088989, "global_step/max_steps": "1205/13541", "percentage": "8.90%", "elapsed_time": "13h 14m 13s", "remaining_time": "5d 15h 30m 47s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025287}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.5625, "completions/min_length": 2.0, "completions/max_length": 5.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.08906285, "global_step/max_steps": "1206/13541", "percentage": "8.91%", "elapsed_time": "13h 14m 48s", "remaining_time": "5d 15h 29m 21s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025289}
{"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1e-06, "reward": -5.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/AntifraudReward/mean": -5.0, "rewards/AntifraudReward/std": 0.0, "completions/mean_length": 2.28125, "completions/min_length": 2.0, "completions/max_length": 5.0, "completions/clipped_ratio": 0.0, "clip_ratio/low_mean": 0.0, "clip_ratio/low_min": 0.0, "clip_ratio/high_mean": 0.0, "clip_ratio/high_max": 0.0, "clip_ratio/region_mean": 0.0, "epoch": 0.0891367, "global_step/max_steps": "1207/13541", "percentage": "8.91%", "elapsed_time": "13h 15m 31s", "remaining_time": "5d 15h 29m 18s", "memory(GiB)": 108.22, "train_speed(iter/s)": 0.025287}
`

**Comment by aJupyter** [comment]
蹲

**Comment by unlabeledData** [comment]
请问您有试过官方提供的CHORD 脚本，在AI-MO--NuminaMath-TIR task是否能够取得效果呢？
我试过官方提供的AI-MO--NuminaMath-TIR任务，训练log显示loss收敛，reward变大，但是lm-eval的测试结果是0 （训练前也是0）.
